### PR TITLE
ADD english version of pulsar IT blog

### PR DESCRIPTION
**Pull request by g-guichard**

### CODE REVIEW

File: Architecture/messaging/pulsar-english.md [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+### Presentation
+
+We will implement an extended Pulsar cluster on 2 regions/datacenters with an active datacenter and a passive that is used only in case of active datacenter failure. In nominal mode, clients consume on the same datacenter as themselves, which reduces latency and the expensive inter-datacenter bandwidth.
+
+1. Pulsar is configured so that the brokers chosen for the partitions of a topic are on a single datacenter.
+2. A client publishes data on this topic.
+3. The data is persistently synchronized on at least one bookie of the active datacenter and at least one bookie of the passive data center. Thus in case of loss of the active datacenter, the data will always be available on the second datacenter without the possibility of losing messages.
+4. During the consumption, the client connects to the broker owner of the topic, this broker will preferentially read the data on a bookie of the same region as him.
+
+In the event of active data center failure, passive data center brokers automatically become usable for publishing/consuming messages. Since there is only one Pulsar cluster, the flip-flop is transparent to the clients.
```
Comment (cbornet) [inline Architecture/messaging/pulsar-english.md]: flip-flop -> failover

File: Architecture/messaging/pulsar-english.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+      --broker-service-url pulsar://pulsar1-eu:6650
+```
+
+Then we create brokers and bookies :
+
+```
+docker-compose -f docker-compose_sync.yml up -d
+```
+
+Then you have to define the regions and racks on which bookies and brokers will be placed thanks to the **_set-bookie-rack_** command.
```
Comment (cbornet) [inline Architecture/messaging/pulsar-english.md]: Then **we** have...

File: Architecture/messaging/pulsar-english.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+```
+docker stop bk1-eu
+```
+
+The bookie is no longer available, the topic is under-replicated. BookKeeper has a [self-healing](https://bookkeeper.apache.org/docs/latest/admin/autorecovery/) mechanism that will automatically replicate the topic data on the new bookie used in the active region **_bk2-eu_** to restore the write quorum. This is why after the loss of **_bk1-eu_** (at 13:44), we observe a peak of writing on **_bk2-eu_** and a peak of reading on **_bk1-us_**.
+![](perteBK1_write.png)
+
+![](pertebk1_read.png)
+
+During the time of self-repair, the client consumes the messages on **_bk1-us_**. Once **_bk2-eu_** replicates the data, the client consumes the messages on it.
```
Comment (cbornet) [inline Architecture/messaging/pulsar-english.md]: self-repair or self-healing ?

File: Architecture/messaging/pulsar-english.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+```
+docker-compose -f docker-compose_geo.yml down
+docker-compose -f docker-compose_zk.yml down
+```
+
+##Realize an active / active messaging bus with strong data consistency guarantee
+As we have seen, Pulsar namespaces and region-awareness features provide strong guarantees for message delivery while minimizing cross-datacenter exchanges with an active/passive cluster. But for our messaging needs, it was important to have active/active replication. To achieve this, we have combined synchronous replication and asynchronous geo-replication. This has several advantages:
+
+- Active/passive synchronous replication ensures that no messages are lost even if a datacenter is lost.
+- When switching to passive nodes, clients do not change cluster and therefore no subscription. There is therefore no need to have a complex mechanism to find the corresponding read index on the other cluster.
```
Comment (cbornet) [inline Architecture/messaging/pulsar-english.md]: > and therefore no subscription

and therefore they keep the same subscription

Comment (cbornet) [reply Architecture/messaging/pulsar-english.md]: Therefore repetition

File: Architecture/messaging/pulsar-english.md [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+- In nominal mode, customers produce and consume on their own region that saves the bandwidth consumed between regions.
+- Geo-replication can receive all messages regardless of the cluster on which they were produced.
+
+However, there are some disadvantages:
+
+- Passive brokers can be considered as a provisioned but unused resource. However, these resources only need to be started permanently if we are looking for maximum availability. If a momentary loss of availability is acceptable in the event of a flip-flop, it is conceivable to start these brokers only when a flip-flop is detected. You can even use paid-for-use resources in the cloud that will only be used during the flip-flop.
```
Comment (cbornet) [inline Architecture/messaging/pulsar-english.md]: flip-flop -> failover

File: Architecture/messaging/pulsar-english.md [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+
+- **Topic** : structure in which messages are published and consumed. In Pulsar or Kafka, topics are persisted so messages do not need to be consumed as soon as they are published and many consumers can read the same messages at different indexes and speeds.
+- **Namespace** : allows to configure the topics policy that it will contain (retention, ACL, persistence, etc.)
+- **Tenant** : Pulsar is multi-tenant, each tenant having its own authentication and authorization scheme. A tenant can contain multiple namespaces.
+
+### Why are we interested in Pulsar?
+
+Pulsar has several features that make it unique compared to other messaging systems:
+
+- **Event delivery but also Message Queue** : By allowing multiple consumer groups to have their own index on the message queue, Pulsar allows event broadcasting uses on the same principle as Kafka. But Pulsar can also validate the processing of messages individually without blocking the message queue (or partition) which is not supported by Kafka and it is essential for use as queing system.
```
Comment (geraudster) [inline Architecture/messaging/pulsar-english.md]: queing -> queuing

File: Architecture/messaging/pulsar-english.md [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+- **_Region Awareness_** : allows messages to be replicated synchronously on bookies from different regions.
+- **_Read reordering_** : allows to privilege the reading of the messages on bookies belonging to the same region as the broker (when the functionality Region Awareness is used)
+
+### Demonstration with Docker
+
+We will build an architecture with 2 datacenters, with 2 brokers and 2 bookies on each one, as well as 1 common ZooKeeper.
+
+The first datacenter represents the region where there will be 2 brokers and 2 bookies which will be prefixed by eu. We will find the same configuration on the second datacenter that represents the us region.
+
+On each region, we create a namespace in active configuration and a another one in passive mode.
```
Comment (geraudster) [inline Architecture/messaging/pulsar-english.md]: a another -> another

File: Architecture/messaging/pulsar-english.md [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+```
+docker-compose -f docker-compose_geo.yml down
+docker-compose -f docker-compose_zk.yml down
+```
+
+##Realize an active / active messaging bus with strong data consistency guarantee
+As we have seen, Pulsar namespaces and region-awareness features provide strong guarantees for message delivery while minimizing cross-datacenter exchanges with an active/passive cluster. But for our messaging needs, it was important to have active/active replication. To achieve this, we have combined synchronous replication and asynchronous geo-replication. This has several advantages:
+
+- Active/passive synchronous replication ensures that no messages are lost even if a datacenter is lost.
+- When switching to passive nodes, clients do not change cluster and therefore no subscription. There is no need to have a complex mechanism to find the corresponding read index on the other cluster.
```
Comment (geraudster) [inline Architecture/messaging/pulsar-english.md]: therefore no subscription -> therefore *their* subscription (?)

File: Architecture/messaging/pulsar-english.md [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+- Geo-replication can receive all messages regardless of the cluster on which they were produced.
+
+However, there are some disadvantages:
+
+- Passive brokers can be considered as a provisioned but unused resource. However, these resources only need to be started permanently if we are looking for maximum availability. If a momentary loss of availability is acceptable in the event of a failover, it is conceivable to start these brokers only when a failover is detected. You can even use paid-for-use resources in the cloud that will only be used during the failover.
+- Since we only validate a message produced when it has been replicated to the other region, this introduces latency to writing. This additional latency may be a brake for some applications and it will then necessary to choose between a very strong consistency of the data and the writing performance.
+
+![](https://raw.githubusercontent.com/Cdiscount/IT-Blog/master/Architecture/messaging/images/activeActive.png)
+
+In case of loss of the EU region, the switch is automatically to the US region:
```
Comment (geraudster) [inline Architecture/messaging/pulsar-english.md]: the switch is *done* automatically

**Pull request closed without merging**